xml-language-server-plugin/src/xml-language-server-backend.ts

```diff
@@ -0,0 +1,49 @@
+
+/**
+ * Generated using theia-plugin-generator
+ */
+
+import * as theia from '@theia/plugin';
+import * as path from 'path';
+
+export function start(context: theia.PluginContext) {
+    const JAR_PATH = path.resolve(__dirname, '..', 'lsp4xml', 'org.eclipse.lsp4xml-all.jar');
+    const XML_LS_ID = 'xml';
+
+    const xmlLanguageServerInfo: theia.LanguageServerInfo = {
+        id: XML_LS_ID,
+        name: 'XML',
+        globPatterns: ['**/*.xml'],
+        command: 'java',
+        args: ['-jar', JAR_PATH]
+    }
+
+    context.subscriptions.push(theia.languageServer.registerLanguageServerProvider(xmlLanguageServerInfo));
+
+    registerCommands(xmlLanguageServerInfo, context);
+}
+
+function registerCommands(lsInfo: theia.LanguageServerInfo, context: theia.PluginContext): void {
+    const startCommand = {
+        id: 'Start XML LS',
+        label: "XML: Start XML Language Server"
+    };
+
+    const stopCommand = {
+        id: 'Stop XML LS',
+        label: "XML: Stop XML Language Server"
+    };
+
+    context.subscriptions.push(
+        theia.commands.registerCommand(startCommand, (...args: any[]) => {
+            context.subscriptions.push(theia.languageServer.registerLanguageServerProvider(lsInfo));
```

Comment:
What if it invoked twice or more? The same about stop.

Reply:
Can you clarify please what do you mean?

Reply:
I would say: if user invoke two times "start LS" command or call "stop" before "start"

Reply:
If user invokes two times start,  the implementation of API will take care of it. There will be message like 'ls with this id is already registered'. 
If user calls stop before start, the API will check that ls with this ID isn't registered and nothing will happen.

Reply:
Thanks for explanation